conans/client/cmd/uploader.py

```diff
@@ -165,11 +167,22 @@ def _prepare_package(self, pref, prev_bundle):
 
     def _compress_package_files(self, layout, pref):
         download_pkg_folder = layout.download_package()
-        package_tgz = os.path.join(download_pkg_folder, PACKAGE_TGZ_NAME)
-        if is_dirty(package_tgz):
-            self._output.warning("%s: Removing %s, marked as dirty" % (str(pref), PACKAGE_TGZ_NAME))
-            os.remove(package_tgz)
-            clean_dirty(package_tgz)
+        compression_format = self._app.cache.new_config.get("core.upload:compression_format") or "gzip"
```

Comment:
is `...new_config` not a dictionary? aka 
```suggestion
        compression_format = self._app.cache.new_config.get("core.upload:compression_format", "gzip")
```